lib/commons/aria/get-role.js

```diff
@@ -0,0 +1,25 @@
+/* global aria, axe */
+
+aria.getRole = function getRole(
+	node,
+	{ noImplicit, tokenList, abstracts, dpub } = {}
+) {
+	const roleAttr = (node.getAttribute('role') || '').trim().toLowerCase();
+	const roleList = tokenList ? axe.utils.tokenList(roleAttr) : [roleAttr];
+
+	// Get the first valid role:
+	const validRoles = roleList.filter(role => {
+		if (!dpub && role.substr(0, 4) === 'doc-') {
+			return false;
+		}
+		return aria.isValidRole(role, { allowAbstract: abstracts });
+	});
+	const explicitRole = validRoles[0];
```

Comment:
So if `validRoles` is an empty array, `explicitRole` will be undefined.
Line 24 below will do a `undefined || null` and `null` will be returned. Is that what is expected? Asking, as in some places we expect an undefined, so long as it is document and we agree on the return, happy with this.

Reply:
Right, tests below, tell me `null` is expected as return. So all good.